### ISSUE TITLE
Add rule to enforce blank line before control blocks (if/for/while/etc.)

### ISSUE THREAD

**Issue by MathieuLoutre** [astral-sh/ruff#18255]
### Summary

I'd like to suggest a rule that flags when a blank line is missing above a control block (if, for, while, try, with etc.) that isn't directly after a `def`. It improves readability especially when a block begins a new logical step or decision in a function.

Input
```python
def process_data(data):
    if data not None:
        cleaned = clean(data)
        if not cleaned:
            return None
```

Output
```python
def process_data(data):
    if data not None:
        cleaned = clean(data)

        if not cleaned:
            return None
```

I'm happy to try to implement this rule and contribute a PR if this seems like a good idea. Let me know where this best belongs and if you have any preferences for configuration or rule naming.

Thanks for all the work on Ruff!

**Comment by MichaReiser** [comment]
Hi. Thanks for the suggestion

This seems similar to https://github.com/astral-sh/ruff/issues/13080 except that it should enforce blank lines before an if statement. 

For now, we decided to not add any new formatting lint rules until we figured out how they'll fit into https://github.com/astral-sh/ruff/issues/1774.
